src/PolykeyAgent.ts

```diff
@@ -455,6 +471,26 @@ class PolykeyAgent {
   }) {
     try {
       this.logger.info(`Starting ${this.constructor.name}`);
+      // Set up error handling for event handlers
+      this.events[captureRejectionSymbol] = (err, event, ...args) => {
+        this.logger.error(
```

Comment:
Have a look at `src/bin/utils/utils.ts` `outputFormatter`:

```
    output += `${msg.name}: ${msg.description}`;
    if (msg.message) {
      output += ` - ${msg.message}`;
    }
    output += '\n';
```

This would produce a message like:

```
ErrorName: Description - Message
```

Now of course since we are in the event bus system, we can add additional context:

```
EventBus error for <event>: <errorname>: <errordescription> - <errormessage>
```

Note that `errordescription` can be swapped for the `errormessage` if it doesn't exist. Because not all errors are Polykey errors. You just have to do an `if` condition like:

```
if (err.description != null) {
  // add description in and message in
  msg += `${err.name}: ${err.description}`
  if (err.message != '') {
    msg += ` - ${err.message}`;
  }
} else {
  // add message in
  msg += `${err.name}`
  if (err.message != '') {
    msg += `: ${err.message}`;
  }
}
```

Maybe there's even slimmer way with just ternary conditions.

Anyway we will be refactoring how errors/log messages are reported in the future in #304, but this is good enough solution for now.

Reply:
So use this for now?

```
if (err.description != null) {
  // add description in and message in
  msg += `${err.name}: ${err.description}`
  if (err.message != '') {
    msg += ` - ${err.message}`;
  }
} else {
  // add message in
  msg += `${err.name}`
  if (err.message != '') {
    msg += `: ${err.message}`;
  }
}
```

Reply:
Oh yea and I forgot to mention, to not show the arguments of the events. This can leak sensitive information to the logs, so we don't do this.

Reply:
> 

Not just that, also make sure that `EventBus ${event}: ` is also there.

Reply:
Try with ternary conditions: `const s = `${(true) ? 'abc': 'what'}`;`.

Reply:
Not sure if this is any clearer but this is that same logic using ternary conditions:
```ts
let output = `EventBus error for ${event}: ${err.name}`;
output +=
  err.description !== null
    ? `: ${err.description}` +
      (err.message !== '' ? ` - ${err.message}` : '')
    : err.message !== ''
    ? `: ${err.message}`
    : '';
```

Reply:
One can use the `if` condition at the top and ternary only when I was adding data on. Actually I just realised we can follow what `src/polykey.ts` does.

```ts
const msg = '${this.constructor.name} error for ${event}';
if (err instanceof ErrorPolykey) {
  msg += `: ${err.name}: ${err.description}`;
  if (err.message != '') {
    msg += ` - ${err.message}`;
  }
} else {
  msg += `: ${err.name}`;
  if (err.message != '') {
    msg += `: ${err.message}`;
  }
}
this.logger.error(msg);
```

Just use the above.

Reply:
Done